tests/unit/store/flask.spec.js

```diff
@@ -340,26 +344,26 @@ describe("store/flask", () => {
         await store.dispatch("flask/start_status_pinging");
       });
 
-      test("When stop_status_pinging is committed, the interval is cleared and the status_ping_interval_id state becomes null", async () => {
+      test("When stop_status_pinging is committed, Then the interval is cleared and the status_ping_interval_id state becomes null", async () => {
         const initial_interval_id = store.state.flask.status_ping_interval_id;
         // confirm pre-condition
         expect(initial_interval_id).toBeGreaterThanOrEqual(0);
 
         const spied_clear_interval = jest.spyOn(window, "clearInterval");
 
         store.commit("flask/stop_status_pinging");
-        expect(spied_clear_interval).toBeCalledWith(initial_interval_id);
-        expect(store.state.flask.status_ping_interval_id).toBe(null);
+        expect(spied_clear_interval).toHaveBeenCalledWith(initial_interval_id);
+        expect(store.state.flask.status_ping_interval_id).toBeNull();
       });
     });
-    describe("ping_system_status plate bar code scanned", () => {
+    describe("SERVER_READY", () => {
       let context = null;
-      let valid_plate_barcode = "MB190440991";
-      let invalid_plate_barcode = "MD20044099";
+      const valid_plate_barcode = "MB190440991";
+      const invalid_plate_barcode = "MD20044099";
       beforeEach(async () => {
         context = await store.dispatch("flask/get_flask_action_context");
       });
-      test("Given the current state is SERVER_READY and the returned state is CALIBRATION_NEEDED with a valid plate bar code value, Then the  Playback State should update to CALIBRATION_NEEDED and the plate_barcode contains the value of SCANNED of barcode value 'MB190440991' obtained from JSON object", async () => {
+      test("Given that the /system_status is mocked with a response of CALIBRATION_NEEDED, When the ping_system_status is active, Then the  Playback State should update to CALIBRATION_NEEDED and the plate_barcode contains the value of SCANNED of barcode value 'MB190440991' obtained from JSON object", async () => {
```

Comment:
we want to avoid including specific values in test titles (`MB190440991`)--that makes the test titles brittle. For example, if we needed to change the barcode format in the future, we would also need to change the title of this test.  Try and make the test title use generic terms to describe what is being tested, rather than including specific numbers/strings.  For example `Given that /system_status is mocked to include a valid plate barcode............Then the barcode value in Vuex is set to the value from the JSON object in /system_status`

Reply:
Corrected

Reply:
Looks like the rules keep changing quite often its so confusing:-
a) Initially it as to have Given, When and Then
b) Now this is a new rule that Barcode not to be put in title.

test.each() definitely has %s which has values of Barcode and strings in the title

Reply:
You are right, it would be better if the `test.each` did not include the specific string of the barcode in its test title.  A test title should try and be descriptive of the aspect of the code being tested, and not include exact numerical or string details of the values being asserted in the test.  The barcode test titles should describe the concept being tested (13 characters, incorrect Julian day format, illegal unicode character, etc) as opposed to having the specific barcode string in the test title.

The goals that we are striving for don't change.  Sometimes when we initially write code it wasn't implemented in the best, cleanest, easiest to read way.  But everytime we refactor we leave the code a little bit cleaner than when we found it.  And as we get more experience writing clean test titles, we get better when we first write the code of making the titles clear, concise, and descriptive for when someone new (or ourselves) come back to read the code later.  Over the course of creating a whole application and maintaining it, 10 times as much time is spent reading code than spent writing it, so it is important to put in the effort to make sure our code is easy to read.